sources/niLang/src/DataTablePath.cpp

```diff
@@ -122,7 +146,25 @@ tBool __stdcall ParseDataTablePathOp(const achar* aaszPath, tDataTablePathOpPtrC
     }
     // DataTable
     else {
-      aLst.push_back(niNew cDataTablePathOp(eDataTablePathOp_DataTable,it->Chars()));
+      cString name = *it;
+      auto itNext = it + 1;
+      auto beginPredicate = *it == _A("[");
+      auto predicateWillBegin = itNext != vToks.end() && *itNext == _A("[");
+      if (!beginPredicate && !predicateWillBegin) {
+        aLst.push_back(niNew cDataTablePathOp(eDataTablePathOp_DataTable, name.Chars()));
+      } else {
+        if (beginPredicate) {
+          name.Clear(); // anonymous datatable name
```

Comment:
I see, that's why `name` isnt const. I think using an optional `astl::optional<const cString*>` would be better for this.

Reply:
Or `astl::optional<const_iterator>` should also work.